Libraries/MiscDrivers/SRAM/aps6404.c

```diff
@@ -29,6 +29,16 @@ enum MODE { STANDARD_MODE, QUAD_MODE };
 typedef enum MODE MODE_t;
 MODE_t g_current_mode;
 
+// Utility macro for validating an error code.  Assumes an 'err' variable
+// of type int exists in the macro's context
+#define ERR_CHECK(x)             \
+    {                            \
+        err = (x);               \
+        if (err != E_NO_ERROR) { \
+            return x;            \
+        }                        \
+    }
+
```

Comment:
This definition is really nice. Moving forward, maybe we can start using this to minimize all the error-if statements that are widespread across our drivers.